src/backend_model_instance.cc

```diff
@@ -558,7 +559,8 @@ TritonModelInstance::PrepareRequestsOrRespond(
   // If any errors occurred, respond with error for each request.
   if (!status.IsOk()) {
     for (auto& r : requests) {
-      InferenceRequest::RespondIfError(r, status, true /* release_requests */);
+      InferenceRequest::RespondIfError(
+          r, status, true /* release_requests */, FailureReason::OTHER);
```

Comment:
Can you think of any test cases to test some of the `reason=OTHER` scenarios (not counting ensembles)? Can be a follow-up, don't need to be in these 24.07 PRs, but would be good to have a test case for these.